### PR TITLE
depricate doi in favor of url

### PR DESCRIPTION
**Pull request by aaronpeikert**
closes https://github.com/formal-methods-mpi/Literature-Research-Pilot/issues/53

### CODE REVIEW

**Comment by aaronpeikert** [comment]
Checking for DOIs only makes sense if every entry would have a doi. However, we require any entry to have at least an url (which is doi.org/theactualdoi for doi-ed entries).

File: src/deprecated.jl [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  dois = []
+     for i in keys(x)
+         push!(dois, location(get(x.records, i, false)).doi)
+     end
+  return dois
+end
+
+@deprecate doi(x) url(x)
+
+function check_doi(x::RecordDatabase, y::Record)
```
Comment (nickhaf) [inline src/deprecated.jl:12]: Just for my understanding: Why would we keep these checks at all? I'd think it makes sense for the extractors, but why the checks?

**Review by nickhaf — APPROVED** [review]
Great, thanks for the update.

File: src/deprecated.jl [hunk 2]
```diff
@@ -0,0 +1,31 @@
+doi(x::Record) = x.location.doi
+function doi(x::RecordDatabase) 
+  dois = []
+     for i in keys(x)
+         push!(dois, location(get(x.records, i, false)).doi)
+     end
+  return dois
+end
+
+@deprecate doi(x) url(x)
```
Comment (nickhaf) [inline src/deprecated.jl:10]: Do we really need to deprecate this? It probably was used only internally?

Comment (aaronpeikert) [reply src/deprecated.jl:10]: I agree that it was only used internally. However, it was an exported function and thus formally requires a depricate.

**Pull request merged**